src/main/java/com/github/andriiyan/sprongtraining/impl/dao/EventDaoImpl.java

```diff
@@ -2,20 +2,54 @@
 
 import com.github.andriiyan.sprongtraining.api.dao.EventDao;
 import com.github.andriiyan.sprongtraining.api.model.Event;
+import com.github.andriiyan.sprongtraining.api.storage.Storage;
+import com.github.andriiyan.sprongtraining.impl.utils.StreamUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
 class EventDaoImpl extends BaseDaoImpl<Event> implements EventDao {
 
     private static final Logger logger = LoggerFactory.getLogger(EventDaoImpl.class);
 
+    public EventDaoImpl(Storage<Event> storage) {
+        super(storage);
+    }
+
     @Override
     public Logger getLogger() {
         return logger;
     }
 
+
+    @Override
+    public List<Event> getEventsByTitle(String title, int pageSize, int pageNum) {
+        final List<Event> events = StreamUtils.paging(findAll()
+                .stream()
+                .filter(event -> event.getTitle().contains(title)), pageNum, pageSize)
+                .collect(Collectors.toList());
+        logger.debug("getEventsByTitle was invoked with title=" + title + ", pageSize=" + pageSize + ", pageNum=" + pageNum
```

Comment:
Please, do not concatenate in logs - use template placeholders. Each concatenation creates a new string in stringpool and in log-heavy applications it may affect performance. Even if you set application log-level to INFO or ERROR - this concatenation in logger.debug will be executed. With templates nothing will happen if log-level is disabled

`logger.debug("getEventsByTitle was invoked with title={}, pageSize={}, pageNum={}, returning {}", title, pageSize, pageNum, events);`
        
Please fix logging in whole apllication

Reply:
Thanks, missed that, done